openedx/features/teams/views.py

```diff
@@ -0,0 +1,267 @@
+from django.http import Http404
+from django.core.urlresolvers import reverse
+from django.shortcuts import render_to_response, redirect
+from django_countries import countries
+from django.conf import settings
+from django.utils.translation import ugettext as _
+from django.contrib.auth.decorators import login_required
+from w3lib.url import add_or_replace_parameter
+
+from django_comment_client.utils import has_discussion_privileges
+from opaque_keys.edx.keys import CourseKey
+from courseware.courses import get_course_with_access, has_access
+from lms.djangoapps.teams.models import CourseTeam, CourseTeamMembership
+from lms.djangoapps.teams import is_feature_enabled
+from nodebb.models import TeamGroupChat
+from student.models import CourseEnrollment, CourseAccessRole
+from lms.djangoapps.teams.serializers import (
+    BulkTeamCountTopicSerializer,
+)
+
+from .helpers import serialize, validate_team_topic, make_embed_url
+from .serializers import CustomCourseTeamSerializer
+
+
+def get_user_recommended_team(course_key, user):
+    user_country = user.profile.country
+    recommended_teams = CourseTeam.objects.filter(course_id=course_key,
+                                                  country=user_country).exclude(users__in=[user]).all()
+
+    return list(recommended_teams)
+
+
+@login_required
+def browse_teams(request, course_id):
+    user = request.user
+    course_key = CourseKey.from_string(course_id)
+    course = get_course_with_access(user, "load", course_key)
+
+    if not is_feature_enabled(course):
+        raise Http404
+
+    if not CourseEnrollment.is_enrolled(user, course.id) and \
+            not has_access(user, 'staff', course, course.id):
+        raise Http404
+
+    # Even though sorting is done outside of the serializer, sort_order needs to be passed
+    # to the serializer so that the paginated results indicate how they were sorted.
+    topics = get_alphabetical_topics(course)
+
+    topics_data = serialize(
+        topics,
+        request,
+        BulkTeamCountTopicSerializer,
+        {'course_id': course.id},
+    )
+
+    recommended_teams = serialize(
+        get_user_recommended_team(course_key, user),
+        request,
+        CustomCourseTeamSerializer,
+        {'expand': ('user',)}
+    )
+
+    context = {
+        'course': course,
+        'topics': topics_data,
+        'recommended_teams': recommended_teams,
+        'user_country': request.user.profile.country.name.format()
+
+    }
+
+    return render_to_response("teams/browse_teams.html", context)
+
+
+@login_required
+def browse_topic_teams(request, course_id, topic_id):
+    course_key = CourseKey.from_string(course_id)
```

Comment:
let's write a helper which takes course_id, user and returns the course with user access. And then use that helper in all views